src/ksm/client/discovery.go

```diff
@@ -216,17 +219,18 @@ func (sd *discoverer) nodeIP() (string, error) {
 
 // NewDiscoverer instantiates a new Discoverer required for discovering node IP
 // of kube-state-metrics pod and endpoint of kube-state-metrics service
-func NewDiscoverer(logger log.Logger, kubernetes client.Kubernetes) client.Discoverer {
-	return NewStaticEndpointDiscoverer("", logger, kubernetes)
+func NewDiscoverer(logger log.Logger, kubernetes client.Kubernetes, namespace string) client.Discoverer {
+	return NewStaticEndpointDiscoverer("", logger, kubernetes, namespace)
 }
```

Comment:
Constructor with optional args triggers me (specially since now there are more than two args). Would it be feasible to add a `Namespace()` method to the object, that would set this internally? Or perhaps as an argument of the `Discover()` method?

Reply:
It's not really optional, `""` just gives you all namespaces and it's a valid value. But yeah, at this point it there could be a config struct for it.

Reply:
What I mean was more like you can construct the object without it, because all namespaces (`""`) is a very reasonable default.

Reply:
Refactored in #172